src/libp2p_transport.erl

```diff
@@ -138,6 +138,31 @@ start_client_session(TID, Addr, Connection) ->
     Handlers = libp2p_config:lookup_connection_handlers(TID),
     case libp2p_multistream_client:negotiate_handler(Handlers, Addr, Connection) of
         {error, Error} -> {error, Error};
+        server_switch ->
+            ChildSpec = #{ id => make_ref(),
+                           start => {libp2p_multistream_server, start_link, [Connection, Handlers, TID]},
+                           restart => temporary,
+                           shutdown => 5000,
+                           type => worker },
+            SessionSup = libp2p_swarm_session_sup:sup(TID),
+            case supervisor:start_child(SessionSup, ChildSpec) of
```

Comment:
This seems like  code that was copied from elsehwere? Any way to have that more common? Where did it come from?

Reply:
Its how the multicast_server is initialized, it's not easy to make it generic, I don't think?